librz/analysis/arch_profile.c

```diff
@@ -0,0 +1,150 @@
+// SPDX-FileCopyrightText: 2021 RizinOrg <info@rizin.re>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_types.h>
+#include <rz_util.h>
+#include <rz_arch.h>
+#include <stdio.h>
+#include <string.h>
+
+/**
+ * \brief Frees an RzArchProfile type
+ *
+ * 	Frees the hashtables used for MMIO and extended
+ * 	registers
+ */
+RZ_API void rz_arch_profile_free(RzArchProfile *p) {
+	if (!p) {
+		return;
+	}
+	ht_up_free(p->registers_mmio);
+	ht_up_free(p->registers_extended);
+}
+
+/**
+ * \brief Creates a new RzArchProfile type
+ */
+RZ_API RZ_OWN RzArchProfile *rz_arch_profile_new() {
+	RzArchProfile *profile = RZ_NEW0(RzArchProfile);
+	profile->registers_mmio = ht_up_new0();
+	profile->registers_extended = ht_up_new0();
+	return profile;
+}
+
+/**
+ * \brief Creates a new RzArchTarget type
+ */
+RZ_API RZ_OWN RzArchTarget *rz_arch_target_new() {
+	RzArchTarget *profile = RZ_NEW0(RzArchTarget);
```

Comment:
verify that profile is not null. same for the pointers below.

Reply:
Actually, I had verified that `profile` is not `NULL` but was told that I don't need to use it since I use `RZ_NEW0` -- https://github.com/rizinorg/rizin/pull/1193#discussion_r646609724

I'll verify for the rest.

Reply:
No, you still need to verify the `profile`, but no need for zero/null-asignment of its members.

Reply:
Oh, alright, sorry I didn't catch that.

Reply:
yes, because if you are on low memory, then malloc/RZ_NEW* will return a null pointer

Reply:
Wow, yeah, that makes sense. #TIL. 
Also, the fact that `RZ_NEW` is basically a `*alloc` just didn't occur to me until now. 😅